numpy/distutils/from_template.py

```diff
@@ -255,10 +252,11 @@ def main():
         (base, ext) = os.path.splitext(file)
         newname = base
         outfile = open(newname, 'w')
-
     allstr = fid.read()
     writestr = process_str(allstr)
     outfile.write(writestr)
+    fid.close()
+    outfile.close()
```

Comment:
Same as above, not ok as this closes stdin/stdout.

Use `contextlib_nullcontext` here too